Sources/Classes/Views/CustomizableButton/CustomizableButtonView.swift

```diff
@@ -0,0 +1,355 @@
+        switch appearance.spinnerPosition {
+        case .center:
+            spinnerView?.centerXAnchor.constraint(equalTo: button.centerXAnchor).isActive = true
+            spinnerView?.centerYAnchor.constraint(equalTo: button.centerYAnchor).isActive = true
+        case .leftToText(let offset):
+            if let buttonLabel = button.titleLabel {
+                spinnerView?.centerYAnchor.constraint(equalTo: buttonLabel.centerYAnchor).isActive = true
+                spinnerView?.trailingAnchor.constraint(equalTo: buttonLabel.leadingAnchor,
+                                                       constant: -offset).isActive = true
+            }
+        case .rightToText(let offset):
+            if let buttonLabel = button.titleLabel {
+                spinnerView?.centerYAnchor.constraint(equalTo: buttonLabel.centerYAnchor).isActive = true
+                spinnerView?.leadingAnchor.constraint(equalTo: buttonLabel.trailingAnchor,
+                                                      constant: offset).isActive = true
+            }
+        }
+    }
+
+    private func configureShadowViewConstraints() {
+        shadowView.constaintToEdges(of: button, with: .zero)
+    }
+}
+
+private extension UIView {
+    func constaintToEdges(of view: UIView, with offset: UIEdgeInsets) {
+        self.translatesAutoresizingMaskIntoConstraints = false
+        self.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: offset.left).isActive = true
+        self.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: offset.right).isActive = true
+        self.topAnchor.constraint(equalTo: view.topAnchor, constant: offset.top).isActive = true
+        self.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: offset.bottom).isActive = true
+    }
+}
+
+extension CustomizableButtonView: InitializableView {
+
+    public func addViews() {
+        addSubviews(shadowView, button)
+    }
+
+    public func configureAppearance() {
+
+        button.titleLabel?.font = appearance.buttonFont
+
+        button.set(titles: appearance.buttonStateTitles)
+        button.set(attributtedTitles: appearance.buttonStateAttributtedTitles)
+        button.set(titleColors: appearance.buttonTitleStateColors)
+        button.set(images: appearance.buttonStateIcons)
+        button.set(backgroundColors: appearance.buttonBackgroundStateColors)
+
+        let offset = appearance.buttonIconOffset
+        button.imageEdgeInsets = UIEdgeInsets(top: offset.vertical,
+                                              left: offset.horizontal,
+                                              bottom: -offset.vertical,
+                                              right: -offset.horizontal)
+
+        if let cornerRadius = appearance.buttonCornerRadius {
+            button.layer.cornerRadius = cornerRadius
+        }
+
+        button.titleLabel?.isHidden = true
+    }
+}
+
+extension CustomizableButtonView: ConfigurableView {
```

Comment:
move inside class declaration to be able override

Reply:
fixed